🎨 UI/UX Improvements: Enhanced Lending Interface with Expandable Details

Bagman365 opened the pull request:
## 🎨 UI/UX Improvements: Enhanced Lending Interface

This PR refactors the lending interface with improved layouts, better information hierarchy, and expandable details sections to enhance user experience when interacting with deposits, borrows, and withdrawals.

## 📋 Changes Overview

### BorrowsList & DepositsList
- **Layout Refactor**: Migrated from flex to grid layout for better responsive behavior
- **Information Hierarchy**: Reorganized asset information display (icon, value, APY, tokens, price) for improved readability
- **Zero Balance Filtering**: Filter out zero-balance borrows from display
- **Button Layout**: Changed action buttons to vertical stack layout for better mobile experience
- **WAD Asset Handling**: Special handling for WAD asset to show "Mint" button instead of "Borrow"

### DepositModal Enhancements
- Added **Total Deposits** display with tooltip
- Added **Market Capacity** display with tooltip
- Added **Estimated APY** section for better transparency

### WithdrawModal Major Improvements
- **Expandable Position Details**: New collapsible section with detailed metrics
- **Enhanced Metrics Display**: 
  - Supply APY
  - Utilization
  - Collateral Factor
  - Accrued Interest
  - Liquidation Margin (optional)
  - Health Factor (optional)
  - LTV (optional)
- **Improved Layout**: Sticky positioning for details card on larger screens

### SupplyBorrowStats Refactor
- Replaced tooltips with expandable details for better UX
- Added **Estimated APY** section with collapsible explanations
- Improved Utilization and Collateral Factor display with expandable details

### Additional Improvements
- Added wallet balance display to `SupplyBorrowForm` for deposit mode
- Updated `MarketsTable` to use custom div styling instead of Card component
- Enhanced `MarketsDesktopTable` styling with improved hover effects and visual hierarchy
- Integrated `MintModal` for WAD asset in `PortfolioModals`
- Added `accruedInterest` to assetData interface
- Minor styling improvements to `MarketsTableActions` and `STokenRow`

## 🎯 Benefits

- **Better Information Architecture**: More organized and scannable layout
- **Improved Mobile Experience**: Vertical button stacks and responsive grid layouts
- **Enhanced Transparency**: Expandable details provide more context without cluttering the UI
- **Consistent Design**: Unified styling and interaction patterns across components

## 🧪 Testing

- [x] Test deposit flow with various assets
- [x] Test borrow flow (including WAD mint flow)
- [x] Test withdraw flow and verify position details display correctly
- [x] Verify responsive behavior on mobile devices
- [x] Test expandable details sections in WithdrawModal and SupplyBorrowStats
- [x] Verify zero-balance borrows are filtered correctly

## 📸 Screenshots

_Add screenshots of the improved UI if available_

## 🔗 Related Issues

_Link any related issues here_

Bagman365 committed:
Your descriptive commit message here

Bagman365 committed:
refactor: improve lending UI with enhanced layouts and expandable details

- Refactor BorrowsList and DepositsList to use grid layout with improved information hierarchy
  - Filter out zero-balance borrows from display
  - Reorganize asset info display (icon, value, APY, tokens, price)
  - Change action buttons to vertical stack layout
  - Add special handling for WAD asset (show "Mint" instead of "Borrow")

- Enhance DepositModal with additional market information
  - Add Total Deposits display
  - Add Market Capacity display
  - Add Estimated APY section

- Major improvements to WithdrawModal
  - Add expandable Position Details section with collapsible metrics
  - Display Supply APY, Utilization, Collateral Factor, and Accrued Interest
  - Add optional fields: Liquidation Margin, Health Factor, and LTV
  - Improve layout with sticky positioning for details card

- Refactor SupplyBorrowStats to use expandable details instead of tooltips
  - Add Estimated APY section with expandable details
  - Improve Utilization and Collateral Factor display with collapsible explanations

- Add wallet balance display to SupplyBorrowForm for deposit mode

- Update MarketsTable to use custom div styling instead of Card component

- Improve MarketsDesktopTable styling and layout
  - Enhance table row hover effects and visual hierarchy
  - Remove info icon button, improve asset display

- Integrate MintModal for WAD asset in PortfolioModals
  - Add accruedInterest to assetData interface

- Minor styling improvements to MarketsTableActions and STokenRow

temptemp3 committed:
Merge branch 'preszi' of github.com:DorkFi/dorkfi-app into preszi

temptemp3 commented:
changes to market table
<img width="1190" height="979" alt="Screenshot 2025-11-27 at 9 15 34" src="https://github.com/user-attachments/assets/3d1737b9-e32a-45f4-8a66-44f95d1af29a" />
before
<img width="1195" height="975" alt="Screenshot 2025-11-27 at 9 15 28" src="https://github.com/user-attachments/assets/e01f9338-6d2e-433b-8b75-8e145f4a3d52" />
after

temptemp3 commented:
changes to deposit modal
<img width="473" height="656" alt="Screenshot 2025-11-27 at 9 16 32" src="https://github.com/user-attachments/assets/45e8f8a9-ac96-4077-b2f8-a3f190f6dd19" />
before
<img width="510" height="725" alt="Screenshot 2025-11-27 at 9 16 27" src="https://github.com/user-attachments/assets/ad5ae141-b7bb-4733-bd6a-4f58047bf872" />
after

temptemp3 commented:
changes to borrow modal
<img width="472" height="663" alt="Screenshot 2025-11-27 at 9 17 36" src="https://github.com/user-attachments/assets/aa88bcf2-742e-44ad-8faa-757f0911ce33" />
before
<img width="470" height="606" alt="Screenshot 2025-11-27 at 9 17 29" src="https://github.com/user-attachments/assets/03b92716-cac7-4368-8c02-33d2fbc7e953" />
after 
---
removes utilisation and cf

temptemp3 commented:
<img width="1183" height="970" alt="Screenshot 2025-11-27 at 9 46 13" src="https://github.com/user-attachments/assets/20e0ae53-9e8b-4160-aa5f-c39605fb2992" />
<img width="1177" height="950" alt="Screenshot 2025-11-27 at 9 45 39" src="https://github.com/user-attachments/assets/7d946bed-e5a1-4d13-a67d-410ca661c316" />
<img width="1181" height="972" alt="Screenshot 2025-11-27 at 9 45 27" src="https://github.com/user-attachments/assets/3356f441-5062-402f-b4ba-e12c1f3ae3fe" />
profile update, layout changes, mint button for wad instead of borrow

temptemp3 committed:
refactor: remove info icons and improve market component styling

- Remove info icon buttons from all market components (MarketCardView,
  MarketsDesktopTable, MarketsTabletTable, STokenCard, STokenRow,
  STokenTabletRow)
- Standardize market icon sizes and add object-contain class for better
  image display consistency
- Add text-muted-foreground styling to CF badges for improved visual
  hierarchy
- Temporarily hide wallet balance display in SupplyBorrowForm deposit
  mode (with TODO to add back)

temptemp3 committed:
Merge branch 'next' into preszi

temptemp3 reviewed (approved):
Lgtm
